### PR TITLE
chore: Only run `deploy_template` after `deploy_image` has been completed

### LINKED ISSUE #7662: condition `deploy_template` job on `deploy_image` job completion
## Problem
When we make changes to https://github.com/coder/coder/blob/main/dogfood/Dockerfile, `dogfood.yaml` action deploys the template `deploy_template` asynchronously to `deploy_image` and the dogfood templates get updated before a newer image is updated so that update does not include the changes made in `Dockerfile`

https://github.com/coder/coder/blob/f47ecc29f0e7e675b2ee1d39b1d856ddb0e76141/.github/workflows/dogfood.yaml#L51
## Solution

We should condition `deploy_template` job on `deploy_image` completion

### PR DESCRIPTION
**Pull request by matifali**
This resolves #7662

### CODE REVIEW

File: .github/workflows/dogfood.yaml [hunk 1]
```diff
@@ -49,6 +49,8 @@ jobs:
           cache-from: type=registry,ref=codercom/oss-dogfood:latest
           cache-to: type=inline
   deploy_template:
+    needs: deploy_image
+    if: ${{ needs.deploy_image.result == 'success' || needs.deploy_image.result == 'skipped' }}
```
Comment (mafredri) [inline .github/workflows/dogfood.yaml:53]: In what situations does the job end up being skipped?

**Review by mafredri — APPROVED** [review]

File: .github/workflows/dogfood.yaml (hunk 1 above)
Comment (matifali) [reply .github/workflows/dogfood.yaml:53]: hmm, when there are no changes in the dogfood directory. but wait that will skip the whole workflow. Good catch.

Comment (matifali) [reply .github/workflows/dogfood.yaml:53]: Should we run this on success only or also on failures?

Comment (mafredri) [reply .github/workflows/dogfood.yaml]: Success only should be fine, I think. Not a big deal to change this later if it behaves differently from what we want 👍

**Pull request merged**